src/github_runner_manager/reactive/consumer.py

```diff
@@ -72,6 +73,30 @@ class JobError(Exception):
     """Raised when a job error occurs."""
 
 
+class QueueError(Exception):
+    """Raised when an error when communicating with the queue occurs."""
+
+
+def get_queue_size(queue_config: QueueConfig) -> int:
+    """Get the size of the message queue.
+
+    Args:
+        queue_config: The configuration for the message queue.
+
+    Returns:
+        The size of the queue.
+
+    Raises:
+        QueueError: If an error when communicating with the queue occurs.
+    """
+    try:
+        with Connection(queue_config.mongodb_uri) as conn:
+            with closing(SimpleQueue(conn, queue_config.queue_name)) as simple_queue:
+                return simple_queue.qsize()
+    except KombuError as exc:
+        raise QueueError("Error when communicating with the queue") from exc
```

Comment:
This error is not currently handled anywhere and will bubble up and show in the logs. It currently won't put the charm into error state though, due to https://matrix.to/#/!xdClnUGkurzjxqiQcN:ubuntu.com/$7C20CUrsJKsJmLtrAFf5GlpgmEhBezt_WQwiObhA_yE?via=ubuntu.com&via=matrix.org&via=matrix.debian.social